### ISSUE TITLE
How much RAM?

### ISSUE THREAD

**Issue by jeffgilb** [mattmcspirit/azurestack#64]
How much RAM is required to run the script successfully? It would be cool to have it kind of broken down for full and then how much each of the RP deployments requires to run. I've got a few ASDK boxes, but some are much less RAM heavy than others!

**Comment by mattmcspirit** [comment]
Good point Jeff - I'll add this to my documentation but here anyway - all VMs that my script deploys are Standard A2's with 3.5GB RA. The MySQL/SQL RPs themselves are fixed by the RP installers at 3.5GB each.

- **MySQL RP** - 2 VMs (RP VM, DB Host VM) = **7GB**
- **SQL Server RP** - 2 VMs (RP VM, DB Host VM) = **7GB**
- **App Service** - 8 VMs (File Server, SQL Host, Front End Scale Set, Shared Worker Tiers (2), Publisher Scale Set, CN0-VM, Management Servers Scale Set) = **28GB**

**Total with all RPs deployed = 42GB in addition to the core running Azure Stack ASDK VMs**

I tend to use a Standard A2 for all of my VMs that the script deploys (apart from the fixes SQL/MySQL RP which i just use their default). I wouldn't want to adjust the database hosts required for the database RPs, however for the App Service, I could probably drop the VM size for the File Server to an A1, and most of the App Service VMs to an A1 also, as per here: https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-app-service-deploy#run-the-app-service-resource-provider-installer, which may save 8-10GB RAM, but not much more I suspect.

What are your thoughts?

**Comment by jeffgilb** [comment]
Ahh that’s not a lot at all really! Very cool!

One issue I’ve come across already is that it doesn’t let you choose or specify which Azure subscription to use for registration. I’ve got two subscriptions, but want to use one specifically (as the other fails). We put in this command in the registration docs to account for this behavior:

Get-AzureRmSubscription -SubscriptionID '<Your Azure Subscription GUID>' | Select-AzureRmSubscription<https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-registration>

Maybe there’s a way to specify the subscriptionID GUID to use for registration when running the script?

**Comment by jeffgilb** [comment]
Never mind, I realized you just need to provide the subscriptionID GUID for the -azureRegSubId parameter.

**Comment by mattmcspirit** [comment]
It creates the registration name dynamically based on “asdkreg-$asdkHostName-$runtime”

So you end up with a pretty unique name that’s easily identifiable by the computer name of the ASDK host you’re running on, and the date/time you register the ASDK, down to the second. Likely unique every time.

**Comment by gijs007** [comment]
My colleague recently made a demo which used almost all the RAM on our 128GB server. 

We had about 5GB of free RAM left and the ASDK was a bit sluggish, if you can reduce the VM's RAM usage by 8 - 10 GB without any consequences that would be great :)

**Comment by mattmcspirit** [comment]
Well the Azure Stack VMs on my freshly deployed ASDK host, show 53GB consumed, however, my ASDK host itself is showing a total of 83GB in use, so I'm not sure where that extra has gone. On a 128GB system, 42GB + 83GB wouldn't fit, but even if the random memory loss was lower, it would still be tight.

I've made some adjustments and am running a test run now.  If all is well, I should be able to reduce 42GB down to 25GB.

I'll share my findings soon.

Thanks

**Comment by gijs007** [comment]
Nice work 👍 

I've noticed the infra VM's use dynamic memory, perhaps it makes sense to use this for the App Service VM's?

**Comment by mattmcspirit** [comment]
Nice try....but the Infra VMs aren't deployed by Azure Stack. They are deployed by the deployment scripts, talking directly to Hyper-V, hence their names are Azs-### and not GUIDs, and they can have dynamic memory enabled etc.

Any VMs deployed by Azure Stack itself, have to use the T-Shirt sizes that Azure/Stack provide, and therefore, no dynamic memory.

Any smaller than what I've used above, and I think things will start to break :)
